doc/source/tune/key-concepts.rst

```diff
@@ -4,112 +4,117 @@
 
-Hyperparameters
----------------
-
-What are *hyperparameters?* And how are they different from *model parameters*?
-
-In supervised learning, we train a model with labeled data so the model can properly identify new data values.
-Everything about the model is defined by a set of parameters, such as the weights in a linear regression. These
-are *model parameters*; they are learned during training.
-
-.. image:: /images/hyper-model-parameters.png
-
-In contrast, the *hyperparameters* define structural details about the kind of model itself, like whether or not
-we are using a linear regression or classification, what architecture is best for a neural network, how many layers, what kind
-of filters, etc. They are defined before training, not learned.
-
-.. image:: /images/hyper-network-params.png
+Search Spaces
+-------------
 
-Other quantities considered *hyperparameters* include learning rates, discount rates, etc. If we want our training
-process and resulting model to work well, we first need to determine the optimal or near-optimal set of *hyperparameters*.
+To optimize your *hyperparameters*, you have to define a *search space*.
+A search space defines valid values for your hyperparameters and can specify
+how these values are sampled (e.g. from a uniform distribution or a normal
+distribution).
 
-How do we determine the optimal *hyperparameters*? The most direct approach is to perform a loop where we pick
-a candidate set of values from some reasonably inclusive list of possible values, train a model, compare the results
-achieved with previous loop iterations, and pick the set that performed best. This process is called
-*Hyperparameter Tuning* or *Optimization* (HPO). And *hyperparameters* are specified over a configured and confined
-search space, collectively defined for each *hyperparameter* in a ``config`` dictionary.
+Tune offers various functions to define search spaces and sampling methods.
+:ref:`You can find the documentation of these search space definitions here <tune-sample-docs>`.
 
-tune.run and Trials
--------------------
+Here's an example covering all search space functions. Again,
+:ref:`here is the full explanation of all these functions <tune-sample-docs>`.
 
-Use :ref:`tune.run <tune-run-ref>` to execute hyperparameter tuning. This function manages your experiment and provides many features such as :ref:`logging <tune-logging>`, :ref:`checkpointing <tune-checkpoint-syncing>`, and :ref:`early stopping <tune-stopping-ref>`.
+.. literalinclude:: doc_code/key_concepts.py
+    :language: python
+    :start-after: __config_start__
+    :end-before: __config_end__
 
-.. code-block:: python
+Trials
+------
 
-    # Pass in a Trainable class or function to tune.run, along with configs
-    tune.run(trainable, config={"a": 2, "b": 4})
+You use :ref:`tune.run <tune-run-ref>` to execute and manage hyperparameter tuning and generate your `trials`.
```

Comment:
can we make ":ref:\`tune.run \<tune-run-ref\>\`" have the same formatting as \`\`tune.run()\`\`?

Reply:
I believe it's not possible, but let me check.

Reply:
not possible :/